File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.java
class Solution {
    public int characterReplacement(String s, int k) {
        HashMap<Character, Integer> hm = new HashMap<Character, Integer>();
        int l = 0;
        int res = 0;
        int max = 1;
        for (int r = 0;r<s.length();r++)
        {
            if (!hm.containsKey(s.charAt(r)))
            {
                hm.put(s.charAt(r), 1);
            }
            else
            {
                hm.put(s.charAt(r), hm.get(s.charAt(r)) + 1);
                max = Math.max(max, hm.get(s.charAt(r)));
            }
            while ((r-l+1) - max > k)
            {
                hm.put(s.charAt(l), hm.get(s.charAt(l)) - 1);
                l++;
            }
            res = Math.max(res, (r-l+1));
        }
        return res;
    }
}

File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.py
class Solution(object):
    def characterReplacement(self, s, k):
        """
        :type s: str
        :type k: int
        :rtype: int
        """
        count = {}
        res = 0
        l = 0
        for h in range(len(s)):
            count[s[h]] = count.get(s[h], 0) + 1

            while (h - l + 1) - max(count.values()) > k:
                count[s[l]] -= 1
                l += 1
            
            res = max(res, h - l + 1)
        return res